src/instance.rs

```diff
@@ -277,16 +234,23 @@ impl Instance for Wasi {
             ));
         }
 
-        let interupt = self.interupt.read().unwrap();
-        let i = interupt.as_ref().ok_or(Error::FailedPrecondition(
+        let lr = self.pidfd.lock().unwrap();
+        let fd = lr.as_ref().ok_or(Error::FailedPrecondition(
             "module is not running".to_string(),
         ))?;
-
-        i.interrupt();
-        Ok(())
+        fd.kill(signal as i32)
     }
 
     fn delete(&self) -> Result<(), Error> {
+        let spec = match load_spec(self.bundle.clone()) {
```

Comment:
`load_spec` result can be saved in Instance internal state.

Reply:
Maybe we can deal with this separately?
We are only calling this twice, once on start and once on delete, but storing this adds more complexity to `fn new` (which is tricky because it can't return an error) or more complexity with lazily-loading the spec.